Source/TrainTrack/Classes/Controller.cs

```diff
@@ -0,0 +1,158 @@
+                        {
+                            Console.WriteLine($"{ stations[j].Name } is the final destination for { train.Name }");
+                            Console.WriteLine($"Arrived at { specificTrainTimeTable[i].ArrivalTime }");
+                            Console.WriteLine();
+                        }
+
+                        if (i == 1)
+                        {
+                            Thread.Sleep(2000);
+                        }
+
+                        while (Departure < Arrival)
+                        {
+                            Console.WriteLine($"{train.Name} says: Choo choo { Departure:HH:mm}");
+                            Departure += addOneMinute;
+                            Arrival.AddMinutes(addOneMinute.Minutes);
+                            Thread.Sleep(200);
+                        }
+                    }
+                }
+
+                Console.WriteLine($"{train.Name} Arrival Time: { Arrival:HH:mm}");
+                Console.WriteLine();
+            }
+
+            Console.WriteLine($"Arrived at { endStation.Name } which is");
+            Console.WriteLine($"{ train.Name } 's final destination.");
+            Console.WriteLine();
+            
+            return this;
+        }
+
+        public IController StartThread1(Train train)
+        {
+            _thread1 = new Thread(() => StartTrain(train));
+            _thread1.Start();
+
+            return this;
+        }
+
+        public IController StartThread2(Train train)
+        {
+            _thread2 = new Thread(() => StartTrain(train));
+            _thread2.Start();
+
+            return this;
+        }
+
+        public IController FollowTimeTable(Train train)
+        {
+            specificTrainTimeTable = times.Where(t => t.TrainID == train.ID).ToList();
+
+            return this;
+        }
+
+        public IController StopAtStations(List<Station> stations, Train train)
+        {
+            if (train.ID == 3)
+            {
+                stopStationsT3 = stations.Where(stations => specificTrainTimeTable.Any(t => t.StationID == stations.ID))
+                                         .OrderByDescending(s => s.ID)
+                                         .ToList();
+            }
+            else
+            {
+                stopStationsT2 = stations.Where(stations => specificTrainTimeTable.Any(t => t.StationID == stations.ID))
+                                         .ToList();
+            }
+            return this;
+        }
+
+        public IController LoadTrainWithPassengers(List<Passenger> passengers)
+        {
+            var travelers = passengers.Where(x => x.ID <= 10);
+
+            foreach (var traveler in travelers)
+            {
+                Console.WriteLine($"{traveler.Name} has stepped onboard");
+                Thread.Sleep(500);
+            }
+            Console.WriteLine("\n\n");
+
+            return this;
+        }
+
```

Comment:
Kan man göra C lite mer abstrakt så metoden tar in även crossingID (int), för att på så sätt inte hårdkoda in crossing id i metoden? Tex

public IController SetCrossing(Enum Status, int iD)
        {
            var crossing = new Crossing()
            {
                ID = iD,
                Status = Status,
                Placement = 20
            };

Reply:
Absolut, det håller jag fullt med om. Fick bli simplare pga tidsbrist, men bra input! :)